code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm

```diff
@@ -201,6 +204,17 @@ GLOBAL_LIST_INIT(wraith_banish_very_short_duration_list, typecacheof(list(
 	var/turf/starting_turf = null
 	var/phase_shift_active = FALSE
 	var/phase_shift_duration_timer_id
+	var/phase_shift_duration_alert_id
+
+/datum/action/xeno_action/phase_shift/can_use_action(silent = FALSE, override_flags)
+	. = ..()
+	var/turf/T = get_turf(owner)
+	for(var/obj/machinery/door/poddoor/timed_late/containment/shutter_check AS in GLOB.wraith_no_incorporeal_pass_shutters)
+		if(locate(shutter_check) in T)
+			if(!silent)
+				to_chat(owner, span_xenowarning("We can't Phase Shift while in the space of warp protected shutters!"))
```

Comment:
Why not ust replace this with 
locate(/obj/machinery/door/poddoor/timed_late/containment) in T

Reply:
actually need to reduce the path to poddoor

Reply:
The problem with using a simple locate is that there are two separate paths that need to be respected

Reply:
What? locates check for children as well

Reply:
/obj/machinery/door/poddoor/timed_late/containment AND
/obj/machinery/door/poddoor/shutters/mainship/selfdestruct

I need to concern the check with these paths specifically; the common root of poddoor is too broad unfortunately.

Reply:
then why are you typecasting the for loop if its going to be typepaths not objects